src/media/ros/ros_reader.cpp

```diff
@@ -518,6 +520,24 @@ namespace librealsense
             data[2] = static_cast<float>(msg->angular_velocity.z);
             LOG_DEBUG("RS2_STREAM_GYRO " << motion_frame);
         }
+        if (stream_id.stream_type == RS2_STREAM_MOTION)
+        {
+            auto data = reinterpret_cast<double*>(motion_frame->data.data());
```

Comment:
Can we cast to `const rs2_combined_motion*` like the writer?

Reply:
Better to use same time for both reading and writing

Reply:
done